packages/medusa/src/services/cart.ts

```diff
@@ -664,9 +679,22 @@ class CartService extends TransactionBaseService {
           : lineItem.quantity
 
         // Confirm inventory or throw error
-        await this.inventoryService_
-          .withTransaction(transactionManager)
-          .confirmInventory(lineItem.variant_id, quantity)
+        if (lineItem.variant_id) {
+          const isCovered =
+            await this.productVariantInventoryService_.confirmInventory(
+              lineItem.variant_id,
+              quantity,
+              { sales_channel_id: cart.sales_channel_id }
```

Comment:
nit: maybe could camelCase this following our discussion on casing in option/context objects